enhancements/baremetal/baremetal-operator-management.md

```diff
@@ -0,0 +1,166 @@
+   framework (similar to some Cluster Version Operator capabilities)
+   for platform specific operators, and the Bare Metal Operator would
+   integrate with this framework.
+2. The Cluster Version Operator can gain the ability to manage
+   operators that are platform-specific, and the Bare Metal Operator
+   would move to being managed by the Cluster Version Operator.
+
+## Motivation
+
+In order to bring up a cluster using the baremetal platform:
+
+1. The installer needs to capture bare metal host information and
+   provisioning configuration for later use by baremetal-operator
+2. Something needs to install the CRDs for these bare metal specific
+   resources created by the installer
+3. Something needs to launch baremetal-operator
+
+Currently (1) is achieved by the installer creating:
+
+* A Provisioning resource manifest
+* Manifests for BareMetalHost resources and their associated secrets
+
+and these manifests are applied by the cluster-bootstrap component
+towards the end of the cluster bootstrapping process.
+
+This resource creation step does not succeed until the step (2)
+completes - i.e. the relevant CRDs are applied - and this is currently
+done by the CVO as it applies the manifests for the MAO.
+
+Finally, (3) happens when the MAO detects that it is running on the
+baremetal infrastructure platform and instantiates the BMO deployment.
+
+There are two problems emerging with this design:
+
+* A sense that launching the BMO is outside of the scope of the MAO,
+  particularly since it does not manage operators on any other
+  platform
+
+* Expanding needs for bare metal specific manifests - for example, new
+  CRDs used to drive new BMO capabilities - to be installed early in
+  the cluster bring-up means introducing yet more bare metal specific
+  concerns into the MAO
+
+Steps (2) and (3) are aspects of cluster bring-up which the CVO is
+clearly well-suited. However, the CVO has no concept of managing
+operators that are specific to an infrastructure platform. Therefore,
+to use the CVO capabilities in this case would mean installing base
+metal CRDs and launching the bare metal operator even on platforms
+where it is not needed.
+
+In contrast, the way (2) and (3) is currently achieved by the MAO is
+not at all generic and results in the MAO having substantial built-in
+knowledge of the bare metal platform. A more generic mechanism where
+it could discover and apply any required manifests from BMO would
+mean the addition of operator management capabilities that look very
+much like the CVO's capabilities.
+
+### Goals
+
+Allow bare metal machine management capabilities to be fully
+encapsulated in the Bare Metal Operator.
+
+Ensure that the Bare Metal Operator is only deployed on the
+`baremetal` platform.
+
+### Non-Goals
+
+### Proposal
+
+A choice should be made between the two options listed above:
+
+1. The Machine API Operator adds a generic framework for
```

Comment:
Why does it have to be generic?  Why can't launching BMO under MAO be exactly like launching an actuator, and then BMO manages the actuator?  Or simply make the bare metal actuator own the responsibility of managing lifecycle of its components?

How can we make "managing sub operators" cheaper by reducing deployment complexity?

Reply:
As above, I'll re-work this to propose bare metal having its own SLO.